admob/integration_test/src/integration_test.cc

```diff
@@ -766,30 +799,34 @@ TEST_F(FirebaseAdMobTest, TestRewardedAdLoadAndShow) {
   options.user_id = "123456";
   rewarded->SetServerSideVerificationOptions(options);
 
-  TestUserEarnedRewardListener user_earned_reward_listener;
-  WaitForCompletion(rewarded->Show(&user_earned_reward_listener), "Show");
+  TestUserEarnedRewardListener earned_reward_listener;
+  WaitForCompletion(rewarded->Show(&earned_reward_listener), "Show");
 
   LogDebug(
       "Wait for the Ad to finish playing, click the ad, return to the ad, "
       "then close the ad to continue.");
 
-  while (
-      full_screen_content_listener.num_on_ad_dismissed_full_screen_content_ ==
-      0) {
+  while (content_listener.num_ad_dismissed() == 0) {
     app_framework::ProcessEvents(1000);
   }
 
   LogDebug("Waiting for a moment to ensure all callbacks are recorded.");
   app_framework::ProcessEvents(2000);
 
-  EXPECT_EQ(full_screen_content_listener.num_on_ad_clicked_, 1);
-  EXPECT_EQ(full_screen_content_listener.num_on_ad_showed_full_screen_content_,
-            1);
-  EXPECT_EQ(full_screen_content_listener.num_on_ad_impression_, 1);
-  EXPECT_EQ(
-      full_screen_content_listener.num_on_ad_dismissed_full_screen_content_, 1);
-  EXPECT_EQ(user_earned_reward_listener.num_on_user_earned_reward_, 1);
-  EXPECT_EQ(paid_event_listener.num_on_paid_event_, 1);
+  EXPECT_EQ(content_listener.num_ad_clicked(), 1);
+  EXPECT_EQ(content_listener.num_ad_showed_content(), 1);
+  EXPECT_EQ(content_listener.num_ad_impressions(), 1);
+  EXPECT_EQ(content_listener.num_ad_dismissed(), 1);
+  EXPECT_EQ(content_listener.num_ad_failed_to_show_content(), 0);
+  EXPECT_EQ(earned_reward_listener.num_earned_rewards(), 1);
+  EXPECT_EQ(paid_event_listener.num_paid_events(), 1);
+
+  // Show the Ad again
+  LogDebug("Attempting to show ad again, checking for correct error result.");
+  WaitForCompletion(rewarded->Show(&earned_reward_listener), "Show");
+  app_framework::ProcessEvents(2000);
+  EXPECT_THAT(content_listener.failure_codes(),
+              testing::ElementsAre(firebase::admob::kAdMobErrorAdAlreadyUsed));
```

Comment:
Typically easier to read if you add a using testing::ElementsAre at the top of the file. Then it reads like English... "expect that failure_codes() elements are ..."

Reply:
Done!